회원가입 페이지 구현

daeunleeeee opened the pull request:
# 🔢 이슈 번호
- #5 

## ⚙ 작업 사항
```markdown
features/auth/presentation/signup/
├── SignUpForm.tsx
└── hooks/
    ├── useNickname.ts
    ├── useEmail.ts
    └── usePassword.ts

pages/signUp.page.tsx
```

- 코드 가독성을 위해 `useNickname`, `useEmail`, `usePassword` 로직 분리
- 훅 내부에서 상태, 유효성, 안내 메시지 관리
- 닉네임 입력
  - 길이 2자 이상 검증, 중복 검사 및 상태별 메세지 표시
- 이메일 입력
  - 이메일 형식 검증, 인증번호 전송/재전송, 인증번호 입력 및 확인, 인증 타이머 관리(만료시 에러 반환)
- 비밀번호 입력
  - 최소 8자리, 대소문자, 숫자, 특수문자 포함 검증, 비밀번호 확인 일치 여부 검증

## 📷 스크린샷
<img width="1000" height="947" alt="스크린샷 2025-09-18 오후 3 59 20" src="https://github.com/user-attachments/assets/88ac757b-1226-4657-b462-5ee574e2682e" />


## 👀 기타
- 리액트 훅 폼으로 리팩토링 예정

daeunleeeee committed:
#5 feat: 회원가입 폼 구현
- SignUpForm.tsx: ui 렌더링, 훅 호출
- hooks 폴더: 닉네임, 이메일, 비밀번호 로직  분리

skrgus6269 commented on src/features/auth/presentation/signup/SignUpForm.tsx:
```diff
@@ -0,0 +1,204 @@
+import { usePassword } from './hooks/usePassword';
+
+export default function SignUpForm() {
+  const { nickname, status: nicknameStatus, isChecked, handleChange: handleNicknameChange, checkAvailability, getMessage: getNicknameMessage } = useNickname();
```
아래 useEmail 처럼 최대 가로 길이를 넘으면 자동으로 줄바꿈 되는 프리티어 설정을 찾아보면 좋을거같아요!
너무 길어서 가독성을 해치는것 같습니다 :)

daeunleeeee replied:
넵 `printWidth` 속성값을 줄여볼게요!

skrgus6269 commented on src/features/auth/presentation/signup/hooks/usePassword.ts:
```diff
@@ -0,0 +1,96 @@
+      setPasswordStatus('default');
+      return;
+    }
+    const hasUpper = /[A-Z]/.test(value);
```
- 비밀번호 형식에 대문자가 한자리 반드시 포함이여야 하는 기획에 대해 한번 더 얘기해보면 좋을거같아요!

daeunleeeee replied:
네엡 알겠습니당

skrgus6269 commented on src/features/auth/presentation/signup/SignUpForm.tsx:
```diff
@@ -0,0 +1,204 @@
+          </div>
+
+          <div className="w-full flex gap-3 pt-4">
+            <button type="button" className="w-full p-3 border border-black-200 text-black-400 rounded hover:bg-white-100 ">
```
- 취소 버튼 클릭 시 홈으로 돌아가는 로직 작성 부탁드립니다 :)

daeunleeeee replied:
네 추가해둘게요!

skrgus6269 reviewed (commented):
- input과 textarea에 status가 빠진거같아요! 
   -  실패 시 border가 색상을 가지도록 만들어주세요!

skrgus6269 commented on src/features/auth/presentation/signup/hooks/useEmail.ts:
```diff
@@ -0,0 +1,188 @@
+ * - 상태별 안내 메시지 반환
+ */
+export const useEmail = () => {
+  const [email, setEmail] = useState('');
+  const [status, setStatus] = useState<InputStatus>('default');
+
+  const [verificationCode, setVerificationCode] = useState('');
+  const [verificationStatus, setVerificationStatus] = useState<InputStatus>('default');
+
+  const [isSent, setIsSent] = useState(false);
+  const [isVerified, setIsVerified] = useState(false);
+  const [timer, setTimer] = useState(0);
+  const [isTimerExpired, setIsTimerExpired] = useState(false);
```
상태들이 hooks에 존재하면 api 관련된 react-query 로직도 hooks에 작성을 하면 되는걸까요??
약간 헷갈리네요

daeunleeeee replied:
음 react-query 같은 api 호출 로직까지 이 훅에 포함시키는 게 맞는지는 저도 고민이 되네요. data, domain, presentation 레이어 구분을 어떻게 할지 다시 논의해보면 깔끔해질 것 같아요..!

daeunleeeee committed:
#5 feat: input, textarea에 status 추가, error 테두리색 적용
- 취소 버튼 클릭 시 홈으로 이동
- 프리티어 `printWidth`값 100으로 수정
- input과 textarea에 status 추가
- 실패 시 border가 색상을 가지도록 수정
- 인증번호 재입력 시 상태 초기화
- 2자 이상 입력 조건 삭제
- 공백만 입력시 중복확인 버튼 비활성화
